nifi-commons/nifi-site-to-site-client/pom.xml

```diff
@@ -61,14 +61,14 @@
             <version>1.10.0-SNAPSHOT</version>
         </dependency>
         <dependency>
-            <groupId>org.apache.httpcomponents</groupId>
-            <artifactId>httpclient</artifactId>
-            <version>4.5.5</version>
+            <groupId>org.apache.httpcomponents.client5</groupId>
+            <artifactId>httpclient5</artifactId>
+            <version>5.0-beta4</version>
```

Comment:
It may be best to move this dependency to the jigsaw profile in this module, so that when built in Java 8 it uses version 4.5.5.  Any thoughts or preferences here?  There are a couple ways to set up the dependencies to accomplish this, but they don't seem very maintenance-friendly.

Reply:
Not totally sure if this would work, but can the dependency use all properties like:

```
        <dependency>
            <groupId>${httpclient.group}</groupId>
            <artifactId>${httpclient.artifact}</artifactId>
            <version>${httpclient.versopm}</version>
        </dependency>
```

The default properties would be:
```
<httpclient.group>org.apache.httpcomponents</httpclient.group> 
<httpclient.artifact>httpclient</httpclient.artifact> 
<httpclient.version>4.5.5</httpclient.version> 
```
and then in the jigsaw profile it would override them to:
```
<httpclient.group>org.apache.httpcomponents.client5</httpclient.group> 
<httpclient.artifact>httpclient5</httpclient.artifact> 
<httpclient.version>5.0-beta4</httpclient.version> 
```

Reply:
That should work, yes.  Good idea!  I didn't think of that one myself, but it should resolve that concern.